go-controller/pkg/util/dpu_annotations.go

```diff
@@ -56,75 +59,119 @@ type DPUConnectionStatus struct {
 	Reason string `json:"Reason,omitempty"`
 }
 
-func (scd *DPUConnectionDetails) FromPodAnnotation(podAnnot map[string]string) error {
-	if annot, ok := podAnnot[DPUConnectionDetailsAnnot]; ok {
-		if err := json.Unmarshal([]byte(annot), scd); err != nil {
-			return fmt.Errorf("failed to unmarshal DPUConnectionDetails. %v", err)
+// MarshalPodDPUConnDetails returns a JSON-formatted annotation describing the pod's DPU connection details
+func MarshalPodDPUConnDetails(pannotations *map[string]string, scd *DPUConnectionDetails, netName string) error {
+	annotations := *pannotations
+	if annotations == nil {
+		annotations = make(map[string]string)
+		*pannotations = annotations
+	}
+	podScds := make(map[string]DPUConnectionDetails)
+	ovnAnnotation, ok := annotations[DPUConnectionDetailsAnnot]
+	if ok {
+		// legacy scd annoation are not of different network
+		if err := json.Unmarshal([]byte(ovnAnnotation), &podScds); err != nil {
+			var legacyScd DPUConnectionDetails
+			if err := json.Unmarshal([]byte(ovnAnnotation), &legacyScd); err == nil {
+				podScds = map[string]DPUConnectionDetails{}
+				podScds[types.DefaultNetworkName] = legacyScd
+			} else {
+				return fmt.Errorf("failed to unmarshal ovn pod annotation %q: %v",
+					ovnAnnotation, err)
+			}
 		}
```

Comment:
Can `UnmarshalPodDPUConnDetails` be reused here?

Reply:
hmm, UnmarshalPodDPUConnDetails  is to get the connDetails of a specific netName, and here we'd need connDetails of all netNames. they can share that part but that part is not much.